src/gap.c

```diff
@@ -1523,6 +1523,23 @@ Obj FuncLOAD_DYN (
     if ( info == 0 )
         ErrorQuit( "call to init function failed", 0L, 0L );
 
+    /* info->type used to be a small integer -- test for this case         */
+    if ((UInt)(info->type) < 4)
+        ErrorQuit("LOAD_DYN: kernel module built for older "
+                  "version of GAP",
+                  0L, 0L);
+
+    /* info->type should start with 'D_' */
+    if ((info->type)[0] != 'D' || (info->type)[1] != '_')
+        ErrorQuit("LOAD_DYN: Invalid kernel module", 0L, 0L);
```

Comment:
Why `0L` instead of just `0`? Yes, a lot of code does that -- but there is no reason for it, and we probably should just search and replace all 0L and 1L in our code base.